resource-managers/kubernetes/core/src/main/scala/org/apache/spark/scheduler/cluster/k8s/ExecutorPodsPollingSnapshotSource.scala

```diff
@@ -67,9 +67,9 @@ private[spark] class ExecutorPodsPollingSnapshotSource(
       if (conf.get(KUBERNETES_EXECUTOR_API_POLLING_WITH_RESOURCE_VERSION)) {
         val list = pods.list(new ListOptionsBuilder().withResourceVersion("0").build())
         val newResourceVersion = UnsignedLong.valueOf(list.getMetadata.getResourceVersion())
-        // Replace only when we receive a monotonically increased resourceVersion
+        // Replace only when we receive a monotonically increased or equal resourceVersion
         // because some K8s API servers may return old(smaller) cached versions in case of HA setup.
-        if (resourceVersion == null || newResourceVersion.compareTo(resourceVersion) > 0) {
+        if (resourceVersion == null || newResourceVersion.compareTo(resourceVersion) >= 0) {
```

Comment:
one question, the resource with equal version will be different?

Reply:
The resource will be the same. We simply invoke `snapshotsStore.replaceSnapshot(list.getItems.asScala.toSeq)` again to make it sure that the driver works in the same way of Spark 3.2 and olders.

Reply:
nvm, got the idea. yea, it looks reasonable.